widget/accordioncontainer.go

```diff
@@ -0,0 +1,335 @@
+	for _, i := range a.items {
+		os = append(os, i)
+	}
+	return &accordionContainerRenderer{
+		container: a,
+		objects:   os,
+	}
+}
+
+type accordionContainerRenderer struct {
+	container *AccordionContainer
+	objects   []fyne.CanvasObject
+}
+
+func (r *accordionContainerRenderer) MinSize() fyne.Size {
+	width := 0
+	height := 0
+	for _, i := range r.container.items {
+		min := i.MinSize()
+		width = fyne.Max(width, min.Width)
+		height += min.Height
+	}
+	if len(r.container.items) > 0 {
+		width += theme.Padding() * 2
+		height += theme.Padding() * 2
+	}
+	return fyne.NewSize(width, height)
+}
+
+func (r *accordionContainerRenderer) Layout(size fyne.Size) {
+	x := theme.Padding()
+	y := theme.Padding()
+	w := size.Width - theme.Padding()*2
+	for _, i := range r.container.items {
+		h := i.MinSize().Height
+		i.Move(fyne.NewPos(x, y))
+		i.Resize(fyne.NewSize(w, h))
+		y += h
+	}
+}
+
+func (r *accordionContainerRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *accordionContainerRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *accordionContainerRenderer) Refresh() {
+	for _, i := range r.container.items {
+		i.Refresh()
+	}
+	canvas.Refresh(r.container)
+}
+
+func (r *accordionContainerRenderer) Destroy() {
+}
+
+var _ fyne.Widget = (*accordionItem)(nil)
+
+type accordionItem struct {
+	BaseWidget
+	container *AccordionContainer
+	header    *accordionItemHeader
+	detail    fyne.CanvasObject
+	open      bool
+}
+
+func (i *accordionItem) setOpen(open bool) {
+	if i.open == open {
+		return
+	}
+	if i.open = open; i.open {
+		i.detail.Show()
+	} else {
+		i.detail.Hide()
+	}
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (i *accordionItem) CreateRenderer() fyne.WidgetRenderer {
+	return &accordionItemRenderer{
+		item: i,
+		objects: []fyne.CanvasObject{
+			i.header,
+			i.detail,
+		},
+	}
+}
+
+type accordionItemRenderer struct {
+	item    *accordionItem
+	objects []fyne.CanvasObject
+}
+
+func (r *accordionItemRenderer) MinSize() fyne.Size {
+	min := r.item.header.MinSize()
+	width := min.Width
+	height := min.Height
+	if r.item.open {
+		min := r.item.detail.MinSize()
+		width = fyne.Max(width, min.Width)
+		height += min.Height
+	}
+	return fyne.NewSize(width, height)
+}
+
+func (r *accordionItemRenderer) Layout(size fyne.Size) {
+	height := r.item.header.MinSize().Height
+	r.item.header.Move(fyne.NewPos(0, 0))
+	r.item.header.Resize(fyne.NewSize(size.Width, height))
+	if r.item.open {
+		r.item.detail.Move(fyne.NewPos(0, height))
+		r.item.detail.Resize(fyne.NewSize(size.Width, r.item.detail.MinSize().Height))
+	}
+}
+
+func (r *accordionItemRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (r *accordionItemRenderer) Objects() []fyne.CanvasObject {
+	return r.objects
+}
+
+func (r *accordionItemRenderer) Refresh() {
+	r.item.header.Refresh()
+	r.item.detail.Refresh()
```

Comment:
I wonder if this should have it's hidden state set based on open, instead of relying on setOpen?
That might help with the initial state issue?

Reply:
Done